### ISSUE TITLE
Fix remote shares access with GVFS 1.48

### ISSUE THREAD

**Issue by x80486** [flathub/org.gimp.GIMP#101]
Reference: https://github.com/flathub/flathub/issues/2180

**Comment by hfiguiere** [comment]
How is it broken?

**Comment by hadess** [comment]
It won't be able to talk to gvfsd, so everything that relies on gvfs won't work as expected.

**Comment by Jehan** [comment]
So all stuff like remote mounts and stuff like this, is that it?

From the linked report, all it takes is for us to add ` "--filesystem=xdg-run/gvfsd"` to the list of permission, right? Then we'll have support for gvfs pre- and post-1.48?

**Comment by hfiguiere** [comment]
My question, is in which case does it fail? I couldn't find one, but it's totally possible that the problem is me here. 

The original issue says not every app needs to do that:

> If your application doesn't need to talk to gvfs via D-Bus, then it doesn't need access to xdg-run/gvfsd.

**Comment by hadess** [comment]
> So all stuff like remote mounts and stuff like this, is that it?

Yes, so pretty much required for an app like GIMP that doesn't use portals but wants to integrate into GNOME.

> From the linked report, all it takes is for us to add ` "--filesystem=xdg-run/gvfsd"` to the list of permission, right? Then we'll have support for gvfs pre- and post-1.48?

Yes.

> The original issue says not every app needs to do that:

If your app doesn't talk to gvfs, then also remove all the org.gtk.vfs D-Bus holes that got punched.

In short, if you use a non-portal file chooser, then you want to punch all the holes necessary for gvfs so you can mount remote shares and disks.

**Comment by Jehan** [comment]
> If your app doesn't talk to gvfs, then also remove all the org.gtk.vfs D-Bus holes that got punched.

Nono it does, since it was added as a response to someone asking for it: (commit)[https://github.com/flathub/org.gimp.GIMP/commit/8739c53b6856021a13e8b4f49ce85d26cd3cbef1] - #4372

Anyway maybe we can't reproduce the issue if we don't use gvfs 1.48 yet. I assume it depends on the gvfs present on host-side, @hadess ? (here I have 1.46 on a Fedora 33)

**Comment by hadess** [comment]
> Anyway maybe we can't reproduce the issue if we don't use gvfs 1.48 yet. I assume it depends on the gvfs present on host-side, @hadess ? (here I have 1.46 on a Fedora 33)

It does, gvfs isn't installed inside the sandbox.

**Comment by Jehan** [comment]
Thanks @hadess.

@hfiguiere If you want to do your first direct commit, be my guest. 😉

**Comment by hfiguiere** [comment]
I see, given that I still run Fedora 33...
